src/main/java/ai/lucidtech/las/sdk/Credentials.java

```diff
@@ -32,39 +40,101 @@ public Credentials(String credentialsPath) {
+        this.clientId = clientId;
+        this.clientSecret = clientSecret;
         this.apiKey = apiKey;
+        this.authEndpoint = authEndpoint;
+        this.apiEndpoint = apiEndpoint;
+    }
+
+    public String getApiKey() {
+        return apiKey;
+    }
+
+    private Map readCredentialsFromEnviron() {
+        Map<String, String> pairs = new HashMap<String, String>();
+        pairs.put("clientId", "LAS_CLIENT_ID");
+        pairs.put("clientSecret", "LAS_CLIENT_SECRET");
+        pairs.put("apiKey", "LAS_API_KEY");
+        pairs.put("authEndpoint", "LAS_AUTH_ENDPOINT");
+        pairs.put("apiEndpoint", "LAS_API_ENDPOINT");
+
+        Map<String, String> result = new HashMap<String, String>();
+
+        for (Map.Entry<String, String> entry : pairs.entrySet()) {
+            result.put(entry.getKey(), System.getenv(entry.getValue()));
+        }
+
+        return result;
     }
 
     private void readCredentials(String credentialsPath) {
         // TODO Read INI files properly
         try(FileInputStream input = new FileInputStream(credentialsPath)) {
             Properties properties = new Properties();
             properties.load(input);
+            Map<String, String> credentialsFromEnviron = this.readCredentialsFromEnviron();
 
-            this.accessKeyId = properties.getProperty("access_key_id");
-            this.secretAccessKey = properties.getProperty("secret_access_key");
+            for (Object prop : properties.entrySet()) {
+
+            }
+
+            this.clientId = properties.getProperty("client_id");
+            this.clientSecret = properties.getProperty("client_secret");
             this.apiKey = properties.getProperty("api_key");
+            this.apiEndpoint = properties.getProperty("api_endpoint");
+            this.authEndpoint = properties.getProperty("auth_endpoint");
         } catch (IOException ex) {
             ex.printStackTrace();
         }
     }
 
-    public String getAccessKeyId() {
-        return this.accessKeyId;
-    }
+    private JSONObject getClientCredentials() throws IOException {
+        URL url = new URL("https://" + authEndpoint + "/token?grant_type=client_credentials");
+        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+        conn.setRequestMethod("POST");
+        conn.setRequestProperty("Content-Type", "application/x-www-form-urlencoded");
+        conn.connect();
+        int status = conn.getResponseCode();
 
-    public String getSecretAccessKey() {
-        return this.secretAccessKey;
+        if (status != 200) {
+            throw new RuntimeException("Failed to fetch access token: HTTP response code " + status);
+        }
+
+        Scanner sc = new Scanner(url.openStream());
+        String response = "";
+
+        while (sc.hasNext()) {
+            response += sc.nextLine();
+        }
+
+        JSONObject jsonResponse = new JSONObject(response);
+
```

Comment:
This looks i a tad "low level". Is it better to use HttpClientBuilder and HttpUriRequest?